app/components/containers/containers.html

```diff
@@ -112,7 +112,7 @@
               <td ng-if="state.displayIP">{{ container.IP ? container.IP : '-' }}</td>
               <td ng-if="applicationState.endpoint.mode.provider === 'DOCKER_SWARM'">{{ container.hostIP }}</td>
               <td>
-                <a ng-if="container.Ports.length > 0" ng-repeat="p in container.Ports" class="image-tag" ng-href="http://{{p.host}}:{{p.public}}" target="_blank">
+                <a ng-if="container.Ports.length > 0" ng-repeat="p in container.Ports" class="image-tag" ng-href="http://{{URLPublish}}:{{p.public}}" target="_blank">
```

Comment:
Instead of assigning `0.0.0.0` to the `URLPublish` if it is not defined in the `EndpointProvider` you should use `p.host` if `URLPublish` is not defined here.